### ISSUE TITLE
Support excuting XCM locally with origin passed in pallet xcm-handler

### ISSUE THREAD

**Issue by tolak** [paritytech/cumulus#314]
@shawntabrizi @shaopengw

In current [implementation](https://github.com/paritytech/cumulus/blob/4b69b8d4bfa113fb3af2da34d1b71313bc43fb0c/xcm-handler/src/lib.rs#L144) if the **destination** is Null, the xcm message would be executed locally first. Here we give **execute_xcm** a **MultiLocation::Null**, everything would be fine except if we'd like to know who send this xcm message during the execution of xcm. In our crosschain transfer scenario for example:

First we would build a xcm based crosschain transfering message like below:

```sh
Xcm::WithdrawAsset {
    assets: vec![MultiAsset::ConcreteFungible {
        id: b"PHA2000".into(),
        amount: 10000000000000.into(),
    }],
    effects: vec![Order::DepositReserveAsset {
        assets: vec![MultiAsset::All],
        dest: MultiLocation::X1(
            Junction::Parachain { id: 5000.into() },
        ),
        effects: vec![Order::DepositAsset {
            assets: vec![MultiAsset::All],
            dest: MultiLocation::X1(Junction::AccountId32 {
                NetworkId::Any ,
                id: [xxxxx],
            }),
        }],
    }],
}
```

then we would execute this xcm message locally(it will first execute locally, and xcm executor would parse **DepositReserveAsset**, this order would send the effect to destination e.g parachain 5000 here, and execute it), however in function [xcm-transactor::withdraw_asset](https://github.com/tolak/phala-blockchain/blob/1f2698cb5650c49017f68dfa2f9f6097a36812c4/pallets/xcm-transactor/src/lib.rs#L162) which implemented TransactAsset trait, we need to parse the account from **location** and withdraw specific amount asset from it, the **location** here comes from the one we passed to **execute_xcm**, if we give it a Null, would case failure.

To solve this problem, we made a [workaround](https://github.com/tolak/phala-blockchain/blob/1f2698cb5650c49017f68dfa2f9f6097a36812c4/pallets/xcm-handler/src/lib.rs#L82) which gives the **origin** to **execute_xcm**, if use sudo, should get origin account from sudo pallet.

Further more, i think we should remove locally xcm execution logic from [send_xcm](https://github.com/tolak/phala-blockchain/blob/1f2698cb5650c49017f68dfa2f9f6097a36812c4/pallets/xcm-handler/src/lib.rs#L183), e.g if destination is Null, we should not think it as locally, instead, if we want execute a xcm message locally, we call T::XcmExecutor::execute_xcm directly.

Those just suggestions about xcm-handler, we hope to have further discussion about this including handle more destination types within send_xcm.

**Comment by shaunxw** [comment]
This issue could be closed now, by #357 .

**Comment by tolak** [comment]
@shaunxw Nice job, close this issue.
